test/gurka/gurka.cc

```diff
@@ -1051,5 +1051,48 @@ void expect_path(const valhalla::Api& result, const std::vector<std::string>& ex
 } // namespace raw
 } // namespace assert
 
+// Helper incident factory function
+valhalla::IncidentsTile::Location
+createIncidentLocation(uint32_t edge_index, float start_offset, float end_offset) {
+  valhalla::IncidentsTile::Location edge;
+  edge.set_edge_index(edge_index);
+  edge.set_start_offset(start_offset);
+  edge.set_end_offset(end_offset);
+  return edge;
+}
+
+std::shared_ptr<test::IncidentsReader>
+setup_graph_with_incidents(const gurka::map& map, const std::vector<IncidentArgs>& incidents) {
```

Comment:
Moved from `test_route_incidents.cc` and refactored to be easier to use without in-depth knowledge of how incidents are implemented.

The `incidents` argument is all that is needed for injecting mocked incidents. The earlier implementation required use of edge-id's returned from this function and then manual insertion of the metadata.